src/provider/provider_os_memory_posix.c

```diff
@@ -106,3 +109,48 @@ void os_strerror(int errnum, char *buf, size_t buflen) {
         LOG_PERR("Retrieving error code description failed");
     }
 }
+
+// open a file
+int os_file_open(const char *path) {
+    if (path == NULL) {
+        LOG_ERR("empty path");
+        return -1;
+    }
+
+    int fd = open(path, O_RDWR);
+    if (fd == -1) {
+        LOG_PERR("cannot open the file: %s", path);
+    }
+
+    return fd;
+}
+
+// open a file or create
+int os_file_open_or_create(const char *path, size_t size) {
+    if (path == NULL) {
+        LOG_ERR("empty path");
+        return -1;
+    }
+
+    int fd = open(path, O_RDWR);
+    if (fd == -1) {
+        LOG_PERR("cannot open the file: %s", path);
+
+        fd = open(path, O_RDWR | O_CREAT | O_EXCL, 0600);
```

Comment:
Do we two separate opens instead one with O_CREAT (which creates files  if it not exists)? If yes we need comment explaining why.

Reply:
There is still two opens calls?

Reply:
It is `os_file_open_or_create_ORIG`. The correct one is below - I forgot to remove the previous version.
Done now.

Reply:
Done

Reply:
Done.

@lplewa ping